### DISCUSSION TITLE
Reduce how often location manager updates

### DISCUSSION THREAD

**Discussion by saroar** [pointfreeco/composable-core-location#32] [Q&A]
I am using this SPM manage all is good I have done 

```swift
extension LocationManager: TestDependencyKey {
    public static let previewValue: Self = .live
    public static let testValue: Self = .failing
}

extension LocationManager: DependencyKey {
    public static let liveValue: Self = .live
}

extension DependencyValues {
    public var locationManager: LocationManager {
        get { self[LocationManager.self] }
        set { self[LocationManager.self] = newValue }
    }
}
```
but I am trying now update user coordinate so I found this solution 
https://stackoverflow.com/questions/55189264/reduce-how-often-location-manager-updates
1st answer so I am trying this via SPM 

```swift
return locationManager.set(
                    activityType: nil,//CLActivityType? = nil,
                    allowsBackgroundLocationUpdates: true,
                    desiredAccuracy: nil,//CLLocationAccuracy? = nil,
                    distanceFilter: 20,//CLLocationDistance? = nil,
                    headingFilter: nil,//CLLocationDegrees? = nil,
                    headingOrientation: nil,// CLDeviceOrientation? = nil,
                    pausesLocationUpdatesAutomatically: false,//Bool? = nil,
                    showsBackgroundLocationIndicator: true //Bool? = nil
                )
                .fireAndForget()
```
my app is crashing 
`Thread 1: "Invalid parameter not satisfying: !stayUp || CLClientIsBackgroundable(internal->fClient) || _CFMZEnabled()"`

**Comment by NFulkerson** [comment]
This issue is often because Background Modes is either disabled entirely or does not have the 'Location Updates' background mode capability enabled. You can enable this in the Signing & Capabilities section of your app target in Xcode. You may also need to provide location usage descriptions in the Info.plist.

If you're not using background location updates, you could also set `allowsBackgroundLocationUpdates` to `false` in the location manager.
